### PR TITLE
Interoperability fork

### PR DESCRIPTION
**Pull request by saratnt**
## Description
Handling hard fork related to contract interoperability feature.

## Jira Ticket
[SDK-1353](https://horizenlabs.atlassian.net/browse/SDK-1353?atlOrigin=eyJpIjoiOTZiMzk2ZDA2MGJkNDc5NGE0YTI0NDY1YmVkYzc1ZDAiLCJwIjoiaiJ9)

## Changes
Added a new fork point for interoperability

## Checks
- [ X] Project Builds
- [ X] Project passes tests and checks
- [ ] Updated documentation accordingly
- [ ] Breaking changes have been correctly tagged and notified
 
## Additional information
The epoch used for the fork point still need to be decided. At the moment a random value is used.

[SDK-1353]: https://horizenlabs.atlassian.net/browse/SDK-1353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala [hunk 1]
```diff
@@ -1,87 +1,844 @@
...
+                    "value": "0x0",
+                    "error": "execution reverted",
+                    "calls": [{
+                      "type": "STATICCALL",
+                      "from": "$nativeCallerAddress",
+                      "to": "${NativeTestContract.contractAddress}",
+                      "gas": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+                      "gasUsed": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+                      "input": "0x$NATIVE_CONTRACT_RETRIEVE_ABI_ID",
+                      "error": "invalid opcode: opcode 0xce not defined"
```
Comment (alsala) [inline sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala:394]: Maybe we could add a comment on why we get this 0xce value. As you explained to me it is a value got from EVM which is trying to apply the call internally. The value depends on this particular UT.

File: sdk/src/test/scala/io/horizen/account/state/ContractInteropTestBase.scala [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+import io.horizen.utils.{BytesUtils, Pair}
 import org.junit.{After, Before}
 
 import java.math.BigInteger
+import java.util
+import scala.jdk.CollectionConverters.seqAsJavaListConverter
 
 abstract class ContractInteropTestBase extends MessageProcessorFixture {
+
+  // TODO the actual fork point is to be confirmed
```
Comment (alsala) [inline sdk/src/test/scala/io/horizen/account/state/ContractInteropTestBase.scala:18]: Actually for UTs we can use a value different from the 'official' one

**Review by alsala — CHANGES_REQUESTED** [review]
LGTM, only some minor remarks

File: sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala (hunk 1 above)
Comment (saratnt) [reply sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala:394]: Done.

**Review by paologalligit — APPROVED** [review]

File: sdk/src/test/scala/io/horizen/account/state/ContractInteropTestBase.scala (hunk 2 above)
Comment (saratnt) [reply sdk/src/test/scala/io/horizen/account/state/ContractInteropTestBase.scala:18]: True. Removed the comment.

File: qa/SidechainTestFramework/account/utils.py [hunk 3]
```diff
@@ -37,9 +37,14 @@ def convertZenniesToZen(valueInZennies):
 WITHDRAWAL_REQ_SMART_CONTRACT_ADDRESS = "0000000000000000000011111111111111111111"
 FORGER_STAKE_SMART_CONTRACT_ADDRESS = "0000000000000000000022222222222222222222"
 CERTIFICATE_KEY_ROTATION_SMART_CONTRACT_ADDRESS = "0000000000000000000044444444444444444444"
+MC_ADDR_OWNERSHIP_SMART_CONTRACT_ADDRESS = "0000000000000000000088888888888888888888"
+PROXY_SMART_CONTRACT_ADDRESS = "00000000000000000000AAAAAAAAAAAAAAAAAAAA"
 #   address used for burning coins
 NULL_ADDRESS = "0000000000000000000000000000000000000000"
 
+# TODO It may change
```
[the remark this answers is not in the document]
Comment (saratnt) [reply qa/SidechainTestFramework/account/utils.py:45]: Done

**Review by alsala — APPROVED** [review]

**Comment by saratnt** [comment]
Merged into st/SDK_1434 branch.

**Pull request closed without merging**